src/Tickets/Commerce/Module.php

```diff
@@ -350,6 +381,30 @@ public function get_attendees_by_id( $post_id, $post_type = null ) {
 
 	}
 
+	/**
+	 * Get attendees by ticket ID.
+	 *
+	 * @since TBD
+	 *
+	 * @param int $ticket_id Ticket ID.
+	 *
+	 * @return array List of attendees.
+	 */
+	protected function get_attendees_by_ticket_id( $ticket_id ) {
+		// Check to see if we already have attendees by ticket id stored.
+		if ( isset( $this->attendees_by_ticket_id[ $ticket_id ] ) ) {
+			$attendees = $this->attendees_by_ticket_id[ $ticket_id ];
+		} else {
+			/** @var Tribe__Tickets__Attendee_Repository $repository */
+			$repository = tec_tc_attendees( $this->orm_provider );
+			$attendees  = $repository->by( 'ticket_id', $ticket_id )->all();
+
+			$this->attendees_by_ticket_id[ $ticket_id ] = $attendees;
+		}
+
+		return $this->get_attendees_from_module( $attendees );
+	}
```

Comment:
This method should live in the Attendees class.

Reply:
@bordoni The `get_attendees_by_ticket_id()` method is actually a protected method that exists in the parent class we're extending here. In `Tribe__Tickets__Tickets`, the `get_attendees_by_ticket_id()` method tries to use `tribe_attendees()` to find the attendee by querying all of the different repositories. I wanted to use `tec_tc_attendees()` and only search the Tickets Commerce repo.

Reply:
Where is that being used? I mean the method: `get_attendees_by_ticket_id`.

I think the code is good, but I want to reduce reliance on the Module class, look at the method `get_price_html` from this class, it's just a bypass to a separate class.

Reply:
Here's where it is defined within `Tribe__Tickets__Tickets`:
https://github.com/the-events-calendar/event-tickets/blob/54b05bd623d88688b9ed8c086d3406c46a5cb22b/src/Tribe/Tickets.php#L874-L879

Here's where it is being used within TC `Module`:
https://github.com/the-events-calendar/event-tickets/blob/54b05bd623d88688b9ed8c086d3406c46a5cb22b/src/Tickets/Commerce/Module.php#L342

The method within `Tribe__Tickets__Tickets` uses `tribe_attendees()`, however there is not a way to only reference TC attendees. It queries all attendee post types. By overriding and using `tec_tc_attendees()` we can narrow it down to only TC attendees.

Reply:
Oh yeah for sure, that is correct. What I am suggesting is the following;

Classes `Tribe__Tickets__Tickets` and any child of it needs to be reduced in complexity at all costs, it is currently why it's so hard to maintain the codebase in Event Tickets.

`\TEC\Tickets\Commerce\Module` currently extends `Tribe__Tickets__Tickets` which is why your code added changes to both. 

What I am suggesting is the following; Move both of the methods you added to a Commerce specific class, like the `\TEC\Tickets\Commerce\Ticket` than make the changes you suggested of routing the Tickets Commerce method to make sure we are not using the wrong provider.

These methods "aliases" need to be kept as simple as possible so we can potentially migrate the codebase out of using them.

If you feel like the Ticket class in the Commerce folder might not be the correct place, feel free to create a new class for Ticket Memoization if we need it.

Reply:
My 2¢
> ...tries to use tribe_attendees() to find the attendee by querying all of the different repositories

`tribe_attendees` [uses the ticket's ORM provider already](https://github.com/the-events-calendar/event-tickets/blob/aedc8a68be32a4706092b62c3558554cdaffa8c2/src/template-tags/tickets.php#L1294)

```
function tribe_attendees( $repository = 'default' ) {...}
```

So I'm not seeing the benefit here. Unless you are addressing a bug? Maybe the bug is that we have a separate `tec_tc_attendees` function instead of adding it to `tribe_attendees` via the filter as we do for all other providers?

That being said, the memoization should be at the "base" (parent/abstract/trait) layer unless there is good reason to do otherwise. We want to minimize the amount of things we need to override in children as much as possible.

Reply:
@Camwyn `tribe_attendees()` attempts to find attendees from all repositories (there isn't an option to only query Tickets Commerce), which creates unnecessary queries. By using `tec_tc_attendees()`, it only queries Tickets Commerce queries. The only time the `get_attendees_by_ticket_id()` method is being used is from within the `TEC\Tickets\Commerce\Module` class, which would only be querying Tickets Commerce attendees.